src/librustdoc/clean/auto_trait.rs

```diff
@@ -99,29 +131,126 @@ impl<'a, 'tcx, 'rcx> AutoTraitFinder<'a, 'tcx, 'rcx> {
         let tcx = self.cx.tcx;
         let generics = self.cx.tcx.generics_of(def_id);
 
+        let ty = self.cx.tcx.type_of(def_id);
+        let mut traits = Vec::new();
+        if self.cx.crate_name != Some("core".to_string()) &&
+           self.cx.access_levels.borrow().is_doc_reachable(def_id) {
+            if let ty::TyAdt(_adt, _) = ty.sty {
```

Comment:
why are we only doing this for adts? wouldn't it be practical to also see impls on tuples, array, slices, ...?

Reply:
True! But the answer lies on @eddyb side here. Or at least I recall him asking me to only perform this on `Adt` items. Maybe I got it wrong?